src/uint/shl.rs

```diff
@@ -5,9 +5,22 @@ use core::ops::{Shl, ShlAssign};
 
 impl<const LIMBS: usize> Uint<LIMBS> {
     /// Computes `self << shift`.
+    ///
+    /// Panics if `shift >= Self::BITS`.
+    pub const fn shl(&self, shift: u32) -> Self {
```

Comment:
Do we even need these methods? Primitive types don't have them.

Reply:
It's what makes [this](https://github.com/RustCrypto/crypto-bigint/pull/430#discussion_r1427312479) possible, i.e. using them in a `const fn` context.

Primitive types are a bit magical in that arithmetic operators like `<<` and `>>` work in const contexts even though for all other types they'd thunk through the `Shl`/`Shr` traits which won't be able to do that until `const impl` is available.